internal/guest/runtime/runc/container.go

```diff
@@ -389,6 +413,22 @@ func (c *container) startProcess(
 		}
 	}
 
+	isLoggingSideCarContainerStr, exists := annotations["io.microsoft.bmc.logging.isLoggingSideCarContainer"]
+	if exists {
+		isLoggingSideCarContainer, _ := strconv.ParseBool(isLoggingSideCarContainerStr)
+		if !isLoggingSideCarContainer {
+			// workload container needs to redirect stdout and stderr to fifo pipe.
+			cmd.Stdout = fifoPipe
+			cmd.Stderr = fifoPipe
+
+			// this is needed to make sure that logging container starts first and starts listening to the fifo pipe.
+			time.Sleep(2 * time.Second)
```

Comment:
for the pipes to work.. reader of the pipe need to be setup before the writer to the pipe.. hence I added this static delay. 
is there any other way to handle this?

Reply:
Writing to the pipe should block if the reader is not up yet. So there shouldn't be any issue here. Did you find a problem through testing?

Reply:
yes it was observed while testing that reader needs to be setup before the writer or things dont work. Hence we wanted to start with this and we can revisit this

Reply:
I think we should change this before merging. Given two container starts are inherently asynchronous from each other, we can't reliably expect a two second wait to ensure that one is started before the other.

What behavior specifically did you observe in testing? I tried a simple Linux program where one instance writes to the pipe, and one instance reads from it. I observed that if the write end is started first, it will write to the pipe's buffer until that fills up, and then block. When the read end is started, it will then read from the buffered data, and once the pipe buffer has space, the write end will unblock again.

Sample program: https://gist.github.com/kevpar/6c4f85352f002077ccd17c4e08f3e6b6

Reply:
you were right.. I reran my old scripts and things seems to work irrespective of the order. Earlier I was kind of running various validations as part of my script and might have misunderstood the failures. Fixed